packages/core/src/utils/mediatorhelpers.ts

```diff
@@ -1,16 +1,20 @@
 import { Agent, MediationRecord } from '@credo-ts/core'
 
 export const isMediatorInvitation = async (agent: Agent, url: string): Promise<boolean> => {
-  const invitation = await agent.oob.parseInvitation(url)
-  if (!invitation) {
+  try {
+    const invitation = await agent.oob.parseInvitation(url)
+    if (!invitation) {
+      return false
+    }
+    if (invitation.goalCode === 'aries.vc.mediate') {
+      agent.config.logger.info(`Invitation is a mediator invitation with goal code: ${invitation.goalCode}`)
+      return true
+    }
+    return false
+  } catch (error) {
+    agent.config.logger.info(`Invitation is not a mediator invitation: ${error}`)
```

Comment:
We need a way to capture the error cause/stacktrace for troubleshooting. co-pilot suggests something like:
```
agent.config.logger.info(
  `Invitation is not a mediator invitation: ${error}\nStack: ${error?.stack ?? 'N/A'}\nCause: ${error?.cause ?? 'N/A'}`
)
```
I am not sure if the logger already handle "error" object type properly and tries to capture/log the cause/stack

Reply:
As per standup:
`agent.config.logger.error(...)`